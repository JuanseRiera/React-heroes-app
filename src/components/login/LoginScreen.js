import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../types/authTypes";

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);
  const login = () => {
    const path = localStorage.getItem("lastPath") || "";
    let action = {
      type: types.login,
      payload: {
        name: "Juanse",
      },
    };
    dispatch(action);
    history.replace(path);
  };
  return (
    <div className="container p-5">
      <h1>LoginScreen</h1>
      <hr />

      <button className="btn btn-info" onClick={login}>
        Login
      </button>
    </div>
  );
};
