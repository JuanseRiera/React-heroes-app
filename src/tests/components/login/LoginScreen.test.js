import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../components/types/authTypes";

describe("Prubas del login screen", () => {
  const historyMock = {
    replace: jest.fn(),
  };

  const values = {
    dispatch: jest.fn(),
  };
  let action = {
    type: types.login,
    payload: {
      name: "Juanse",
    },
  };
  const wrapper = mount(
    <AuthContext.Provider value={values}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );
  test("Debe de cargar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de realizar el dispatch y la navegaciÃ³n", () => {
    wrapper.find("button").simulate("click");
    expect(values.dispatch).toHaveBeenCalled();
    expect(values.dispatch).toHaveBeenCalledWith(action);
    expect(historyMock.replace).toHaveBeenCalled();
    expect(historyMock.replace).toHaveBeenCalledWith("");

    localStorage.setItem("lastPath", "/dc");
    wrapper.find("button").simulate("click");
    expect(historyMock.replace).toHaveBeenCalledWith("/dc");
  });
});
