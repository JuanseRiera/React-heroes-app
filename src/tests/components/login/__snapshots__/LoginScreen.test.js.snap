// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas del login screen Debe de cargar correctamente 1`] = `ReactWrapper {}`;
